src/server/Server.cpp

```diff
@@ -135,12 +157,40 @@ void Server::setSocketNonblock() {
 	}
 }
 
-void Server::bindSocketName() {
+void Server::_bindSocketName() {
 	if (bind(_main_socketfd, (struct sockaddr *)&_address, sizeof(_address)) < 0) {
 		throw InitialisationException(strerror(errno));
 	}
 }
 
+std::string Server::_saveFile(const std::string &file_name)
+{
+	std::time_t now = std::time(nullptr);
+	std::tm* now_tm = std::localtime(&now);
+
+	std::ostringstream oss;
+	oss << (now_tm->tm_year + 1900)
+        << (now_tm->tm_mon + 1)
+        << now_tm->tm_mday
+        << now_tm->tm_hour
+        << now_tm->tm_min
+        << now_tm->tm_sec;
+
+	std::string new_file_name;
+	new_file_name += _config.root;
+	new_file_name += "/";
+	new_file_name += "uploads/";
+	new_file_name += oss.str();
+	new_file_name += "-";
+	oss.clear();
+	oss << rand() % 1000;
+	new_file_name += oss.str();
+	new_file_name += ".file";
+
+	rename(file_name.c_str(), new_file_name.c_str());
+	return new_file_name;
```

Comment:
rename may be not allowed

Reply:
is this from a particular specification?

Reply:
subject allows us only certain functions from C, rename is not part of it
as well as remove